shiny/ui/_navs_dynamic.py

```diff
@@ -0,0 +1,163 @@
+        `target=None`, then `"before"` means the new nav item should be inserted at
+        the head of the navlist, and `"after"` is the end.
+    select
+        Whether the nav item should be selected upon insertion.
+    session
+        A :class:`~shiny.Session` instance. If not provided, it is inferred via
+        :func:`~shiny.session.get_current_session`.
+
+    See Also
+    --------
+    ~remove_nav_panel
+    ~update_nav_panel
+    ~shiny.ui.nav_panel
+    """
+
+    session = require_active_session(session)
+
+    # N.B. this is only sensible if the target is a menu, but we don't know that,
+    # which could cause confusion of we decide to support top-level strings at some
+    # in the future.
+    if isinstance(nav_panel, str):
+        nav = menu_string_as_nav(nav_panel)
+    else:
+        nav = nav_panel
+
+    # N.B. shiny.js' is smart enough to know how to add active classes and href/id attrs
+    li_tag, div_tag = nav.resolve(
+        selected=None, context=dict(tabsetid="tsid", index="id")
+    )
+
+    msg = {
+        "inputId": resolve_id(id),
+        "liTag": session._process_ui(li_tag),
+        "divTag": session._process_ui(div_tag),
+        "menuName": None,
+        "target": target,
+        "position": position,
+        "select": select,
+    }
+
+    session._send_message_sync({"shiny-insert-tab": msg})
+
+
+def remove_nav_panel(id: str, target: str, session: Optional[Session] = None) -> None:
+    """
+    Remove a nav item from a navigation container.
+
+    Parameters
+    ----------
+    id
+        The `id` of the relevant navigation container (i.e., `navset_*()` object).
+    target
+        The `value` of an existing :func:`shiny.ui.nav_panel` item to remove.
+    session
+        A :class:`~shiny.Session` instance. If not provided, it is inferred via
+        :func:`~shiny.session.get_current_session`.
+
+    See Also
+    --------
+    ~insert_nav_panel
+    ~update_nav_panel
+    ~shiny.ui.nav_panel
+    """
+
+    session = require_active_session(session)
+    msg = {
+        "inputId": resolve_id(id),
+        "target": target,
+    }
+
+    session._send_message_sync({"shiny-remove-tab": msg})
+
+
+@add_example()
+def update_nav_panel(
+    id: str,
+    target: str,
+    method: Literal["show", "hide"],
+    session: Optional[Session] = None,
+) -> None:
+    """
+    Show/hide a navigation item
+
+    Parameters
+    ----------
+    id
+        The `id` of the relevant navigation container (i.e., `navset_*()` object).
+    target
+        The `value` of an existing :func:`shiny.ui.nav` item to show.
+    method
+        The action to perform on the nav_panel (`"show"` or `"hide"`).
+    session
+        A :class:`~shiny.Session` instance. If not provided, it is inferred via
+        :func:`~shiny.session.get_current_session`.
+
+    Note
+    ----
+    On reveal, the `nav_panel` will not be the active tab. To change the active tab, use `~update_navs()`
```

Comment:
Have you confirmed that the ``~update_navs()``  syntax will work? If yes, let's use that syntax consistently